nemoguardrails/rails/llm/injections.py

```
@@ -0,0 +1,179 @@
+        self._compile_patterns()
+
+    def _compile_patterns(self) -> None:
+        """Compile regex patterns for faster matching."""
+        self.compiled_patterns = []
+        for pattern, name in self.INJECTION_PATTERNS:
+            flags = re.IGNORECASE | re.MULTILINE
+            try:
+                compiled = re.compile(pattern, flags)
+                self.compiled_patterns.append((compiled, name))
+            except re.error as e:
+                raise ValueError(f"Invalid regex pattern '{pattern}': {e}")
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)